### PR TITLE
Adds a new entry to the resx and update all rules

### LINKED ISSUE #1175: Rule S1118 (Utility classes should not have public constructors) ignores its exceptions
### Description

Rule S1118 mentions the exception that classes containing `public static void main(String[] args)` will be ignored. This is not the case.

Original text from Rule description:
> Exceptions
> When class contains `public static void main(String[] args)` method it is not considered as utility class and will be ignored by this rule.

### Repro steps

The following code shows up as a code smell caused by S1118.

```csharp
public class Program
{
    public static void Main(string[] args)
    {
    }
}
```

### Related information

* SonarQube Version: 7.0.0.36138
* SonarC# Version: 6.8.1 (build 4648)
* Visual Studio Version: 15.5.6
* SonarQube Scanner for MSBuild Version: 4.0.2

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #1175

### CODE REVIEW

File: sonaranalyzer-dotnet/rspec/vbnet/S2178_vb.net.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <p>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are
-evaluated under the wrong circumstances.</p>
+evaluated under the wrong circumstances. </p>
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/rspec/vbnet/S2178_vb.net.html]: Trailing whitespace. Does it matter?

File: scripts/rspec/rspec.ps1 [hunk 2]
```diff
@@ -201,11 +207,12 @@ function CreateStringResources($lang, $rules) {
         }
     }
 
+    [void]$resources.Add("HelpLinkFormat=https://rules.sonarsource.com/$($helpLanguageMap.Get_Item($lang))/RSPEC-{0}")
+    [void]$resources.Add("Language=$($roslynLanguageMap.Get_Item($lang))")
```
Comment (valhristov) [inline scripts/rspec/rspec.ps1]: probably RoslynLanguage is better, just Language could be confused with localization language...

File: sonaranalyzer-dotnet/rspec/cs/S3168_c#.html [hunk 3]
```diff
@@ -46,4 +46,7 @@ <h2>Compliant Solution</h2>
   }
 }
 </pre>
+<h2>Exceptions</h2>
```
Comment (valhristov) [inline sonaranalyzer-dotnet/rspec/cs/S3168_c#.html]: Are all of these exceptions actually implemented?

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S2178_vb.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <p>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are
-evaluated under the wrong circumstances.</p>
+evaluated under the wrong circumstances. </p>
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S2178_vb.html]: Trailing whitespace.

File: sonaranalyzer-dotnet/rspec/cs/S3366_c#.html [hunk 5]
```diff
@@ -22,6 +22,9 @@ <h2>Noncompliant Code Example</h2>
   }
 }
 </pre>
+<h2>Exceptions</h2>
```
Comment (valhristov) [inline sonaranalyzer-dotnet/rspec/cs/S3366_c#.html]: Is this implemented?

File: sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/RspecStrings.resx [hunk 6]
```diff
@@ -850,7 +856,7 @@
     <value>Blocker Code Smell</value>
   </data>
   <data name="S2178_Description" xml:space="preserve">
-    <value>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are evaluated under the wrong circumstances.</value>
+    <value>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are evaluated under the wrong circumstances. </value>
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/RspecStrings.resx]: Whitespace.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S3366.html [hunk 7]
```diff
@@ -22,6 +22,9 @@ <h2>Noncompliant Code Example</h2>
   }
 }
 </pre>
+<h2>Exceptions</h2>
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S3366.html]: Is this implemented?

File: sonaranalyzer-dotnet/rspec/cs/S3168_c#.html (hunk 3 above)
Comment (michalb-sonar) [reply sonaranalyzer-dotnet/rspec/cs/S3168_c#.html]: Yes, I've checked.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S3366.html (hunk 7 above)
Comment (Evangelink) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S3366.html]: If you read the updated tests you will see that I have checked this is the case.

**Pull request merged**